burn-import/src/onnx/dim_inference.rs

```diff
@@ -270,13 +280,31 @@ fn slice_update_outputs(node: &mut Node) {
         })
         .unwrap();
 
-    node.outputs[0].ty = ArgType::Tensor(TensorArg { dim: tensor.dim });
+    node.outputs[0].ty = ArgType::Tensor(tensor.clone());
 }
 
 fn same_as_input(node: &mut Node) {
     node.outputs[0].ty = node.inputs[0].ty.clone();
 }
 
+fn equal_update_outputs(node: &mut Node) {
+    let input1_type = node.inputs[0].ty.clone();
+
+    match input1_type {
+        ArgType::Tensor(tensor) => {
+            // if the input is a tensor, the output is a tensor of bool
+            node.outputs[0].ty = ArgType::Tensor(Tensor {
+                elem_type: ElementType::Bool,
+                ..tensor
+            });
+        }
+        ArgType::Scalar(_) => {
+            node.outputs[0].ty = ArgType::Scalar(ElementType::Bool);
+        }
+        _ => panic!("Only tensor input is valid"),
```

Comment:
Looking at the code, it seems like Scalar is also accepted?

Reply:
yes, it's incorrect message. It should say only scalar is accepted for now.